lms/djangoapps/course_structure_api/v0/views.py

```diff
@@ -0,0 +1,236 @@
+""" API implementation for course-oriented interactions. """
+
+import logging
+
+from django.conf import settings
+from django.http import Http404
+from rest_framework.authentication import OAuth2Authentication, SessionAuthentication
+from rest_framework.exceptions import PermissionDenied
+from rest_framework.generics import RetrieveAPIView, ListAPIView
+from xmodule.modulestore.django import modulestore
+from opaque_keys.edx.keys import CourseKey
+
+from courseware.access import has_access
+from openedx.core.djangoapps.content.course_structures import models
+from openedx.core.lib.api.permissions import IsAuthenticatedOrDebug
+from openedx.core.lib.api.serializers import PaginationSerializer
+from courseware import courses
+from course_structure_api.v0 import serializers
+from student.roles import CourseInstructorRole, CourseStaffRole
+
+
+log = logging.getLogger(__name__)
+
+
+class AuthMixin(object):
+    """
+    Mixin that provides authentication and authorization helpers for APIView classes.
+    """
+    authentication_classes = (SessionAuthentication, OAuth2Authentication,)
+    permission_classes = (IsAuthenticatedOrDebug,)
+
+    def user_can_access_course(self, user, course):
+        """
+        Determines if the user is staff or an instructor for the course.
+        Always returns True if DEBUG mode is enabled.
+        """
+        return settings.DEBUG or has_access(user, CourseStaffRole.ROLE, course) or has_access(user,
```

Comment:
The only other place that does something similar is one of the OAuth scope handlers. These permissions are pretty specific to this API, so they shouldn't necessarily be shared.